src/shader_recompiler/frontend/translate/data_share.cpp

```diff
@@ -110,6 +116,42 @@ void Translator::DS_WRITE(int bit_size, bool is_signed, bool is_pair, const GcnI
     }
 }
 
+void Translator::DS_ADD_U32(const GcnInst& inst) {
+    const IR::U32 addr{GetSrc(inst.src[0])};
+    const IR::U32 data{GetSrc(inst.src[1])};
+    const IR::U32 offset = ir.Imm32(u32(inst.control.ds.offset0));
+    const IR::U32 addr_offset = ir.IAdd(addr, offset);
+
+    const IR::U32 value = ir.SharedAtomicIAdd(addr_offset, data);
+    ir.WriteShared(32, value, addr_offset);
```

Comment:
The WriteShared call is not needed as OpAtomicIAdd will write the result as well. The value returned is the original anyway so this is undoing the operation